sdk/dnsresolver/Microsoft.Azure.Management.DnsResolver/tests/ScenarioTests/DnsResolverTests.cs

```diff
@@ -0,0 +1,643 @@
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var createdDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName, tags: null);
+            var tagsForUpdate = TestDataGenerator.GenerateTags();
+
+            var updatedDnsResolver = this.DnsResolverManagementClient.DnsResolvers.Update(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                parameters: TestDataGenerator.GenerateDnsResolver(tags: tagsForUpdate));
+
+            updatedDnsResolver.Should().NotBeNull();
+            updatedDnsResolver.ProvisioningState.Should().BeEquivalentTo(Constants.ProvisioningStateSucceeded);
+            updatedDnsResolver.Tags.Should().BeEquivalentTo(tagsForUpdate);
+        }
+
+        [Fact]
+        public void PatchDnsResolver_ResolverExistsChangeTags_ExpectSuccess()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var createdDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName, tags: TestDataGenerator.GenerateTags());
+            var tagsForUpdate = TestDataGenerator.GenerateTags();
+
+            var updatedDnsResolver = this.DnsResolverManagementClient.DnsResolvers.Update(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                parameters: TestDataGenerator.GenerateDnsResolver(tags: tagsForUpdate));
+
+            updatedDnsResolver.Should().NotBeNull();
+            updatedDnsResolver.ProvisioningState.Should().BeEquivalentTo(Constants.ProvisioningStateSucceeded);
+            updatedDnsResolver.Tags.Should().BeEquivalentTo(tagsForUpdate);
+        }
+
+        [Fact]
+        public void PatchDnsResolver_ResolverExistsRemoveTags_ExpectSuccess()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var createdDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName, tags: TestDataGenerator.GenerateTags());
+
+            var updatedDnsResolver = this.DnsResolverManagementClient.DnsResolvers.Update(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                parameters: TestDataGenerator.GenerateDnsResolver(tags: null));
+
+            updatedDnsResolver.Should().NotBeNull();
+            updatedDnsResolver.ProvisioningState.Should().BeEquivalentTo(Constants.ProvisioningStateSucceeded);
+            updatedDnsResolver.Tags.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void DeleteDnsResolver_ResolverNotExists_ExpectNoError()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var dnsResolverName = TestDataGenerator.GenerateDnsResolverName();
+
+            Action deleteDnsResolverAction = () => this.DnsResolverManagementClient.DnsResolvers.Delete(
```

Comment:
Delete (and patch) don't take IfMatch/IfNoneMatch like Put does?

Looking at DnsResolverService.cs, they should.

Reply:
We only supports IfMatch for delete and IfMatch case added. 
![image](https://user-images.githubusercontent.com/12420143/107006230-b14dc380-6745-11eb-8254-114f6fb994f6.png)